src/btree/bt_sync.c

```diff
@@ -221,8 +227,46 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 				leaf_bytes += page->memory_footprint;
 				++leaf_pages;
 			}
+			++page_counter;
```

Comment:
The other page counters are uint64, is it possible that we can overflow this one being uint32. It doesn't get reset to 0 after every 5000, if timer is false.

Reply:
Moved the page counter increment statement under if(timer) condition.